### PR TITLE
implement legal entity/site/address output routes

### PR DESCRIPTION
**Pull request by pazepaze**

### CODE REVIEW

**Comment by pazepaze** [comment]
gate-output-legal-entities
gate-output-sites

**Review by nicoprow — APPROVED** [review]

File: bpdm-gate/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/controller/AddressController.kt [hunk 1]
```diff
@@ -103,34 +102,19 @@ class AddressController(
...
+        description = "Get page of addresses. Can optionally be filtered by external ids."
     )
     @ApiResponses(
         value = [
             ApiResponse(responseCode = "200", description = "The requested page of addresses"),
             ApiResponse(responseCode = "400", description = "On malformed pagination request", content = [Content()]),
         ]
     )
-    @GetMapping("/output/addresses")
+    @PostMapping("/output/addresses")
```
Comment (nicoprow) [inline bpdm-gate/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/controller/AddressController.kt:113]: better POST output/addresses/search?

File: bpdm-gate/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/controller/LegalEntityController.kt [hunk 2]
```diff
@@ -99,34 +98,19 @@ class LegalEntityController(
...
+        description = "Get page of legal entities. Can optionally be filtered by external ids."
     )
     @ApiResponses(
         value = [
             ApiResponse(responseCode = "200", description = "The requested page of legal entities"),
             ApiResponse(responseCode = "400", description = "On malformed pagination request", content = [Content()]),
         ]
     )
-    @GetMapping("/output/legal-entities")
+    @PostMapping("/output/legal-entities")
```
[the remark this answers is not in the document]
Comment (pazepaze) [reply bpdm-gate/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/controller/LegalEntityController.kt]: done

File: bpdm-gate/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/controller/AddressController.kt (hunk 1 above)
Comment (pazepaze) [reply bpdm-gate/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/controller/AddressController.kt]: done

File: bpdm-gate/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/controller/SiteController.kt [hunk 3]
```diff
@@ -98,34 +97,19 @@ class SiteController(
...
+        description = "Get page of sites. Can optionally be filtered by external ids."
     )
     @ApiResponses(
         value = [
             ApiResponse(responseCode = "200", description = "The requested page of sites"),
             ApiResponse(responseCode = "400", description = "On malformed pagination request", content = [Content()]),
         ]
     )
-    @GetMapping("/output/sites")
+    @PostMapping("/output/sites")
```
[the remark this answers is not in the document]
Comment (pazepaze) [reply bpdm-gate/src/main/kotlin/org/eclipse/tractusx/bpdm/gate/controller/SiteController.kt]: done

**Pull request merged**